add 8 new packages

incebellipipo opened the pull request:
add

-  libmongoc-1.0-0
-  libmongoc-dev
-  libcap2
-  libcap-ng0
-  libcap-ng-dev
-  libprocps6
-  libprocps-dev
-  libdpkg-dev

incebellipipo committed:
add 8 new packages

tfoote reviewed (changes requested):
Please link to the upstream packages for contributed rosdep rules: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md to make it easier to review this and have a record of the intent and scope.

incebellipipo committed:
remove libcap2 from gentoo

incebellipipo committed:
remove libprocps6 from gentoo

libprocps6 is versions (2.3>) in debian and ubuntu package repositories differs from gentoo (3.3>)

incebellipipo commented:
- removed libcap2 from gentoo
- removed libprocps6 from gentoo

|Package|Debian|Gentoo|Ubuntu|
|---|---|---|---|
|libmongoc-1.0-0|[libmongoc-1.0-0](https://packages.debian.org/buster/libmongoc-1.0-0)| |[libmongoc-1.0-0](https://packages.ubuntu.com/bionic/libmongoc-1.0-0)|
|libmongoc-dev|[libmongoc-dev](https://packages.debian.org/buster/libmongoc-dev)|[dev-libs/mongo-c-driver](https://packages.gentoo.org/packages/dev-libs/mongo-c-driver)|[libmongoc-dev](https://packages.ubuntu.com/bionic/libmongoc-dev)|
|libcap2|[libcap2](https://packages.debian.org/buster/libcap2)||[libcap2](https://packages.ubuntu.com/bionic/libcap2)|
|libcap-ng0|[libcap-ng0](https://packages.debian.org/buster/libcap-ng0)|[sys-libs/libcap-ng](https://packages.gentoo.org/packages/sys-libs/libcap-ng)|[libcap-ng0](https://packages.ubuntu.com/bionic/libcap-ng0)|
|libcap-ng-dev|[libcap-ng-dev](https://packages.debian.org/buster/libcap-ng-dev)| |[libcap-ng-dev](https://packages.ubuntu.com/bionic/libcap-ng-dev)|
|libprocps6|[libprocps6](https://packages.debian.org/stretch/libprocps6)| |[libprocps6](https://packages.ubuntu.com/bionic/libprocps6)|
|libprocps-dev|[libprocps-dev](https://packages.debian.org/buster/libprocps-dev)| |[libprocps-dev](https://packages.ubuntu.com/bionic/libprocps-dev)|
|libdpkg-dev|[libdpkg-dev](https://packages.debian.org/buster/libdpkg-dev)| |[libdpkg-dev](https://packages.ubuntu.com/bionic/libdpkg-dev)|

nuclearsandwich commented on rosdep/base.yaml:
```diff
@@ -3232,6 +3252,12 @@ libpqxx-dev:
+  ubuntu: [libprocps-dev]
+libprocps6:
+  debian: [libprocps6]
+  ubuntu: [libprocps6]
```
libprocps6 is only available in Debian Stretch and Ubuntu Bionic respectively. Other versions of this library appear to be used on earlier and later platforms. You'll need to at least scope this key to the distributions where it is available. I'm not sure if we should consider a `libprocps` key that maps to the available version for each distribution.

incebellipipo replied:
You are right, we shouldn't make it available for each distribution since they literally doesn't exist in other versions. I made fix such that make libprocps6 only available to stretch and bionic respectively in 5bc928b.

incebellipipo replied:
Updated comment https://github.com/ros/rosdistro/pull/24685#issuecomment-622704138 accordingly.

nuclearsandwich reviewed (changes requested):
Thanks for adding the requested info. There is an issue with one of the keys (line commented).

incebellipipo committed:
make libprocps6 release specific

incebellipipo commented:
hellö,
Is there any update on review?

clalancette commented on rosdep/base.yaml:
```diff
@@ -6195,3 +6223,4 @@ zziplib:
   opensuse: [zziplib-devel]
   rhel: [zziplib-devel]
   ubuntu: [libzzip-0-13, libzzip-dev]
+
```
Really minor nit, but please remove this trailing whitespace.  I'm surprised it didn't trip our linters.

incebellipipo replied:
oh, sorry. I remove the trailing whitespace.

clalancette reviewed (approved):
One nit to fix, but looks good to me.

incebellipipo committed:
remove trailing spaces

incebellipipo commented:
@clalancette I'm checking the linter script locally, it seems to be not properly functioning for detecting empty lines, but it correctly identifies trailing whitespaces.
Can't detect this:
```yaml
a:
  b: precious



```
Can detect this (if you try to mark the right side of the 'b' you'll see the whitespace:
```yaml
a:
  b: value  
```
I'm working on another PR for `check_rosdep.py` script for detecting empty lines which may save mare mortals like me from embarrassing situations like this :D.

incebellipipo commented:
Adding this will do.
```python
def no_empty_lines(buf):
    clean = True 
    for i, l in enumerate(buf.split('\n')[:-1]):
        if re.match(r'^\s*$', l):
            print_err("empty line %u" % (i+1))
            clean = False
    return clean
```
Results:
```
checking for trailing spaces...
checking for empty lines...
  ERR: empty line 6225
checking for incorrect indentation...
checking for non-bracket package lists...
checking for item order...
building yaml dict...
there were errors, please correct the file
```
I also checked for a case where an empty line also has trailing whitespaces which is kinda rare. If that happens script returns this:
```bash
$ ./check_rosdep.py ../rosdep/base.yaml
checking for trailing spaces...
  ERR: trailing space line 6225
checking for empty lines...
  ERR: empty line 6225
checking for incorrect indentation...
  ERR: line 6224:   
Traceback (most recent call last):
  File "./check_rosdep.py", line 223, in <module>
    if not main(args.infile):
  File "./check_rosdep.py", line 181, in main
    my_assert(correct_indent(buf))
  File "./check_rosdep.py", line 116, in correct_indent
    return generic_parser(buf, fun)
  File "./check_rosdep.py", line 82, in generic_parser
    s = re.search(r'(?!' + indent_atom + ')[^\s]', l).start()
AttributeError: 'NoneType' object has no attribute 'start'
$ echo $?
1
```

I didn't want to commit my local change, since it will automatically be a part of this PR which labelled as `rosdep`. If its ok, I can commit it, if its not I'll wait for this PR to be merged and create a new PR.

clalancette commented:
>  If its ok, I can commit it, if its not I'll wait for this PR to be merged and create a new PR.

Please submit a separate PR for this change, since it isn't directly tied to these rosdep rules.  And thanks for doing that!
